test/test_base.py

```diff
@@ -42,6 +46,18 @@ def setup_logging(self):
         self.log = logging.getLogger("test")
         self.log.info("Logging started")
 
+    def print_and_assert_msgs(self, message):
+        if (self.log_expected_msgs or self.log_unexpected_msgs) and assert_log(
+            message, self.log_expected_msgs, self.log_unexpected_msgs
+        ):
+            self.log_msg_assertions_passed = True
```

Comment:
Does this flag need to get reset ever? What if a test uses assert_log a few times with the same test_base instance?

Reply:
I imagined users setting `self.log_expected_msgs` and `self.log_unexpected_msgs` once per TestBase instance.

Reply:
We could reset the assertions status after checking. I updated with that concept. Need to test it.